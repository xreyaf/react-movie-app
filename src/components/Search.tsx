import React, { ChangeEventHandler } from 'react';

// @ts-ignore
import * as Unicons from '@iconscout/react-unicons';
import styled from '@emotion/styled';

const InputWrapper = styled.div`
  position: relative;
  border-radius: 1rem;
  margin-bottom: 3rem;

  label {
    position: absolute;
    top: 1.75rem;
    left: 4rem;
    font-size: 1.3rem;
    color: ${(props) => props.theme.colors.grey600};
    cursor: text;
    user-select: none;
    transition: all 0.3s cubic-bezier(0.165, 0.84, 0.44, 1);
  }
`;

const StyledInput = styled.input`
  background: transparent;
  background: ${(props) => props.theme.colors.black10};
  z-index: 3;
  width: 100%;
  height: 5rem;
  padding: 1rem 1.3rem 0.3rem 3.9rem;
  font-size: 1.3rem;
  border-radius: 1rem;
  color: ${(props) => props.theme.colors.grey400};
  border: 3px solid ${(props) => props.theme.colors.grey800};
  outline: none;

  &:focus {
    border: 3px solid ${(props) => props.theme.colors.primary500};
    & + label {
      transform: translateY(-1.2rem);
      font-size: 1rem;
    }
    &::placeholder {
      opacity: 0.5;
      transition: all 0.2s;
    }
  }

  &::placeholder {
    opacity: 0;
  }

  &:not(:placeholder-shown) + label {
    transform: translateY(-1.2rem);
    font-size: 1rem;
  }
`;

const IconWrapper = styled.div`
  display: inline-block;
  position: absolute;
  top: 1.8rem;
  left: 1.5rem;
  color: ${(props) => props.theme.colors.grey600};
`;

const Search = ({
  onChange,
  label,
  placeholder,
}: {
  onChange: ChangeEventHandler<HTMLInputElement>;
  label: string;
  placeholder: string;
}) => {
  return (
    <InputWrapper>
      <IconWrapper>
        <Unicons.UilSearch size={24} />
      </IconWrapper>
      <StyledInput
        name="search"
        id="search"
        type="text"
        placeholder={placeholder}
        onChange={onChange}
      />
      <label htmlFor="search">{label}</label>
    </InputWrapper>
  );
};

export default Search;
