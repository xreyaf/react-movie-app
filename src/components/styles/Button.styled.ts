import styled from "@emotion/styled";
import { theme } from "../../styles/theme";
import { motion } from "framer-motion";

export const ButtonStyled = styled(motion.button)`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0.5rem;
  outline: none;
  border: none;
  border-radius: 1rem;
  background: ${theme.colors.primary400};
  color: ${theme.colors.white100};
  padding: 1rem 2rem;
  font-size: 1rem;
  margin-bottom: .6rem;
  svg{
    align-self: baseline;
  }
`;
