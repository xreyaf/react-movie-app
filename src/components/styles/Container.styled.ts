import styled from "@emotion/styled";

export const Container = styled.div`
  width: 100%;
  max-width: 1440px;
  padding: 1rem 5rem;
  margin: 0 auto;
`;
