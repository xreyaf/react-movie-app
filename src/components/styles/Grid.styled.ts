import styled from "@emotion/styled";
import { motion } from "framer-motion";

export const GridStyled = styled(motion.div)`
  display: grid;
  grid-template-columns: repeat(4, minmax(0, 1fr));
  row-gap: 1.4rem;
  column-gap: 1.4rem;
  align-items: start;
  @media screen and (max-width: 1199.98px) {
    grid-template-columns: repeat(3, minmax(0, 1fr));
    transition: all .5s;
  }
  @media screen and (max-width: 991.98px) {
    row-gap: .8rem;
    column-gap: .8rem;
    transition: all .5s;
  }
  @media screen and (max-width: 816.98px) {
    grid-template-columns: repeat(2, minmax(0, 1fr));
    row-gap: .8rem;
    column-gap: .8rem;
    transition: all .5s;
  }
  @media screen and (max-width: 575.98px) {
    grid-template-columns: repeat(1, minmax(0, 1fr));
    transition: all .5s;
  }
`;
