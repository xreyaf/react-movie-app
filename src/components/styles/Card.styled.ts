import styled from "@emotion/styled";
import { theme } from "../../styles/theme";
import { motion } from "framer-motion";

export const CardStyled = styled(motion.div)`
  position: relative;
  border-radius: 1rem;
  overflow: hidden;
  background-color: ${theme.colors.grey900_90};
  width: 100%;
  min-height: 100%;
`;
export const CardImageWrapper = styled.div`
  display: inline-block;
  border-radius: 1rem;
  width: 100%;
  height: 88%;
  overflow: hidden;
`;

export const ImageStyled = styled(motion.img)`
  object-fit: cover;
  width: 100%;
  min-height: 100%;
  vertical-align: middle;
`;

export const CardInfo = styled.div`
  padding: .5rem;
  margin-bottom: .5rem;
  overflow: hidden;

  a {
    color: ${theme.colors.grey50};
    font-weight: 600;

    &:hover {
      color: ${theme.colors.white100};
      transition: color .3s;
    }
  }
`;