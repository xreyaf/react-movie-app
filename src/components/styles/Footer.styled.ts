import styled from "@emotion/styled";
import { theme } from "../../styles/theme";

export const FooterStyled = styled.footer`
  background: ${theme.colors.grey900_80};
  backdrop-filter: blur(2px);
  padding: 5rem 0;
  text-align: center;
`;
