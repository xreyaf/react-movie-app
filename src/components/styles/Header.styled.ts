import styled from "@emotion/styled";
import { theme } from "../../styles/theme";
import { motion } from "framer-motion";

export const HeaderStyled = styled.header`
  position: fixed;
width: 100%;
height: 10%;
  top: 0;
  left: 0;


  z-index: 100;
  background: ${theme.colors.grey900_80};
  backdrop-filter: blur(24px);

  

`;

export const Nav = styled.nav`
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 3rem;

  ul {
    display: flex;
    flex-direction: row;
    justify-content: end;
    align-items: center;
  }
`;

export const Logo = styled(motion.img)`
  max-width: 3rem;
`;

export const StyledLinkMenu = styled(motion.a)`
  display: inline-flex;
  gap: 0.5rem;
  align-items: center;
  padding: 1rem 1rem;
  font-weight: 700;
  font-size: 1.2rem;
  cursor: pointer;
  color: ${theme.colors.grey400};

  &:hover {
    color: ${theme.colors.primary400};
  }
`;
