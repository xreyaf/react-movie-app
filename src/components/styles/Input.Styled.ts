import styled from "@emotion/styled";
import { theme } from "../../styles/theme";

export const InputStyled = styled.div`
  position: relative;
  border-radius: 1rem;
  margin-bottom: 3rem;

  label {
    position: absolute;
    top: 2rem;
    left: 4rem;
    font-size: 1.5rem;
    color: ${theme.colors.grey600};
    cursor: text;
    user-select: none;
    transition: all .3s cubic-bezier(0.165, 0.84, 0.44, 1);

  }
`;

export const InputContent = styled.input`
  background: transparent;
  background: ${theme.colors.black10};
  z-index: 3;
  width: 100%;
  height: 5rem;
  padding: 2rem 1.3rem 1rem 3.9rem;
  font-size: 1.3rem;
  border-radius: 1rem;
  color: ${theme.colors.grey400};
  border: 3px solid ${theme.colors.grey800};
  outline: none;

  &:focus {
    border-color: ${theme.colors.primary500};

    & + label {
      transform: translateY(-1.2rem);
      font-size: 1rem;
    }

    &::placeholder {
      opacity: 0.5;
      transition: all .2s
    }

  }

  &::placeholder {
    opacity: 0;
  }

  &:not(:placeholder-shown) + label {
    transform: translateY(-1.2rem);
    font-size: 1rem;
  }
`;

export const IconWrapper = styled.div`
  display: inline-block;
  position: absolute;
  top: 2rem;
  left: 1.5rem;
  color: ${theme.colors.grey600};
`;