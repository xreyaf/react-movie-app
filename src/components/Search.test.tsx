import { render, screen } from '@testing-library/react';
import Search from './Search';
import { theme } from '../styles/theme';
import { ThemeProvider } from '@emotion/react';
import userEvent from '@testing-library/user-event';
import { matchers } from '@emotion/jest';

expect.extend(matchers);
const onChange = jest.fn();

describe('Search component', () => {
  it('renders Search', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Search
          onChange={onChange}
          placeholder="placeholder"
          label="Search movies or TV series"
        />
      </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should implement onChange', () => {
    render(
      <ThemeProvider theme={theme}>
        <Search
          onChange={onChange}
          placeholder="placeholder"
          label="Search movies or TV series"
        />
      </ThemeProvider>
    );
    const input = screen.getByLabelText(/Search movies or TV series/i);
    userEvent.type(input, 'spider');
    expect(onChange).toHaveBeenCalledTimes(6);
  });
  it('should change border while focus', () => {
    render(
      <ThemeProvider theme={theme}>
        <Search
          onChange={onChange}
          placeholder="placeholder"
          label="Search movies or TV series"
        />
      </ThemeProvider>
    );
    const input = screen.getByLabelText(/Search movies or TV series/i);
    expect(input).toHaveStyleRule(
      'border',
      '3px solid hsla(224, 31.9%, 18.4%, 1)'
    );
    expect(input).toHaveStyleRule(
      'border',
      '3px solid hsla(246, 88.5%, 62.4%, 1)',
      { target: ':focus' }
    );
  });
});
