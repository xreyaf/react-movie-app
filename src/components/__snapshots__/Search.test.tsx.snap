// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders Search 1`] = `
.emotion-0 {
  position: relative;
  border-radius: 1rem;
  margin-bottom: 3rem;
}

.emotion-0 label {
  position: absolute;
  top: 1.75rem;
  left: 4rem;
  font-size: 1.3rem;
  color: hsla(224, 19.3%, 34.5%, 1);
  cursor: text;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.3s cubic-bezier(0.165, 0.84, 0.44, 1);
  transition: all 0.3s cubic-bezier(0.165, 0.84, 0.44, 1);
}

.emotion-1 {
  display: inline-block;
  position: absolute;
  top: 1.8rem;
  left: 1.5rem;
  color: hsla(224, 19.3%, 34.5%, 1);
}

.emotion-2 {
  background: transparent;
  background: hsla(0, 0%, 0%, 0.1);
  z-index: 3;
  width: 100%;
  height: 5rem;
  padding: 1rem 1.3rem 0.3rem 3.9rem;
  font-size: 1.3rem;
  border-radius: 1rem;
  color: hsla(224, 12.3%, 52.2%, 1);
  border: 3px solid hsla(224, 31.9%, 18.4%, 1);
  outline: none;
}

.emotion-2:focus {
  border: 3px solid hsla(246, 88.5%, 62.4%, 1);
}

.emotion-2:focus+label {
  -webkit-transform: translateY(-1.2rem);
  -moz-transform: translateY(-1.2rem);
  -ms-transform: translateY(-1.2rem);
  transform: translateY(-1.2rem);
  font-size: 1rem;
}

.emotion-2:focus::-webkit-input-placeholder {
  opacity: 0.5;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.emotion-2:focus::-moz-placeholder {
  opacity: 0.5;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.emotion-2:focus:-ms-input-placeholder {
  opacity: 0.5;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.emotion-2:focus::placeholder {
  opacity: 0.5;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.emotion-2::-webkit-input-placeholder {
  opacity: 0;
}

.emotion-2::-moz-placeholder {
  opacity: 0;
}

.emotion-2:-ms-input-placeholder {
  opacity: 0;
}

.emotion-2::placeholder {
  opacity: 0;
}

.emotion-2:not(:placeholder-shown)+label {
  -webkit-transform: translateY(-1.2rem);
  -moz-transform: translateY(-1.2rem);
  -ms-transform: translateY(-1.2rem);
  transform: translateY(-1.2rem);
  font-size: 1rem;
}

<div
  class="emotion-0"
>
  <div
    class="emotion-1"
  >
    <svg
      fill="currentColor"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M21.71,20.29,18,16.61A9,9,0,1,0,16.61,18l3.68,3.68a1,1,0,0,0,1.42,0A1,1,0,0,0,21.71,20.29ZM11,18a7,7,0,1,1,7-7A7,7,0,0,1,11,18Z"
      />
    </svg>
  </div>
  <input
    class="emotion-2"
    id="search"
    name="search"
    placeholder="placeholder"
    type="text"
  />
  <label
    for="search"
  >
    Search movies or TV series
  </label>
</div>
`;
